phoenix-core/src/main/java/org/apache/phoenix/schema/types/S3BasedLobStore.java

```diff
@@ -0,0 +1,69 @@
+package org.apache.phoenix.schema.types;
+
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.model.S3Object;
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.RandomStringUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+
+public class S3BasedLobStore implements LobStore {
+
+    private static final String ROOT_DIRECTORY = "/tmp/";
+    private AmazonS3 s3client;
+    private static String BUCKET_NAME = "phoenixgiants-bucket";
+    private static S3BasedLobStore INSTANCE = null;
+
+    public synchronized static S3BasedLobStore getInstance() {
+        if(INSTANCE == null){
+            INSTANCE = new S3BasedLobStore();
+        }
+        return INSTANCE;
+    }
+
+    public S3BasedLobStore() {
+        BasicAWSCredentials credentials = new BasicAWSCredentials("AKIAIRFVDBF3ZVSZDP4Q", "PPLkr8QCRWlPWBYzJQeR2NkJGImg/NVupAD4tGjf");
```

Comment:
I can figure out a way to inject aws credential differently if needed.

Reply:
Reading credentials from resource file/different class for it would've been nice, but +1 for hackathon-level.

Reply:
Thats safer as anyone can look at github and log in using these... probably should at least remove from a public repository like github open source.

Reply:
For our demo we can hard code this though but i'd censor the actual credentials.